plugins/inputs/passenger/passenger.go

```diff
@@ -0,0 +1,217 @@
+	Process_group_id      string `xml:"process_group_id"`
+}
+
+func (p *process) getUptime() int64 {
+	if p.Uptime == "" {
+		return 0
+	}
+
+	timeSlice := strings.Split(p.Uptime, " ")
+	var uptime int64
+	uptime = 0
+	for _, v := range timeSlice {
+		switch {
+		case strings.HasSuffix(v, "d"):
+			iValue := strings.TrimSuffix(v, "d")
+			value, err := strconv.ParseInt(iValue, 10, 64)
+			if err == nil {
+				uptime += value * (24 * 60 * 60)
+			}
+		case strings.HasSuffix(v, "h"):
+			iValue := strings.TrimSuffix(v, "y")
+			value, err := strconv.ParseInt(iValue, 10, 64)
+			if err == nil {
+				uptime += value * (60 * 60)
+			}
+		case strings.HasSuffix(v, "m"):
+			iValue := strings.TrimSuffix(v, "m")
+			value, err := strconv.ParseInt(iValue, 10, 64)
+			if err == nil {
+				uptime += value * 60
+			}
+		case strings.HasSuffix(v, "s"):
+			iValue := strings.TrimSuffix(v, "s")
+			value, err := strconv.ParseInt(iValue, 10, 64)
+			if err == nil {
+				uptime += value
+			}
+		}
+	}
+
+	return uptime
+}
+
+var sampleConfig = `
+	# Path of passenger-status.
+	#
+	# Plugin gather metric via parsing XML output of passenger-status
+	# More information about the tool:
+	#   https://www.phusionpassenger.com/library/admin/apache/overall_status_report.html
+	#
+	#
+	# If no path is specified, then the plugin simply execute passenger-status
+	# hopefully it can be found in your PATH
+	command = "passenger-status -v --show=xml"
+`
+
+func (r *passenger) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *passenger) Description() string {
+	return "Read metrics of passenger using passenger-status"
+}
+
+func (g *passenger) Gather(acc inputs.Accumulator) error {
+	if g.Command == "" {
+		g.Command = "passenger-status -v --show=xml"
+	}
+
+	out, err := exec.Command(g.Command).Output()
+
+	if err != nil {
+		return err
+	}
+
+	if err = importMetric(out, acc); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func importMetric(stat []byte, acc inputs.Accumulator) error {
+	var p info
+
+	decoder := xml.NewDecoder(bytes.NewReader(stat))
+	decoder.CharsetReader = charset.NewReaderLabel
+	if err := decoder.Decode(&p); err != nil {
+		return fmt.Errorf("Cannot parse input with error: %v\n", err)
+	}
+
+	tags := map[string]string{
+		"passenger_version": p.Passenger_version,
+	}
+	fields := map[string]interface{}{
+		"process_count":      p.Process_count,
+		"max":                p.Max,
+		"capacity_used":      p.Capacity_used,
+		"get_wait_list_size": p.Get_wait_list_size,
+	}
+	acc.AddFields("passenger", fields, tags)
+
+	for _, sg := range p.Supergroups.Supergroup {
+		tags := map[string]string{
+			"name": sg.Name,
+		}
+		fields := map[string]interface{}{
+			"get_wait_list_size": sg.Get_wait_list_size,
+			"capacity_used":      sg.Capacity_used,
+		}
+		acc.AddFields("passenger", fields, tags)
+
+		for _, group := range sg.Group {
+			for _, process := range group.Processes.Process {
+				tags := map[string]string{
```

Comment:
Add them, but I also retain the old tags(code_revision, process_gruoup_id, life_status, pid) because it allows user to filter a process for debuging. Like follow a particular `code_revision` that suddenly slowdown the app, or memory leak on a `process`.